### PR TITLE
[PF-388] Rework Sam Exceptions. 

### PR DESCRIPTION
**Pull request by tnadler**
Create a family of Sam Exceptions derived from the corresponding ErrorReportExceptions so they are not checked. Move to a factory model for creating this exception type.

### CODE REVIEW

File: src/main/java/bio/terra/common/sam/exception/SamBadRequestException.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package bio.terra.common.sam.exception;
+
+import bio.terra.common.exception.BadRequestException;
+
+public class SamBadRequestException extends BadRequestException {
```
Comment (wchamber) [inline src/main/java/bio/terra/common/sam/exception/SamBadRequestException.java:9]: nit: Worth some very short javadoc comment on what these Sam exceptions should be used for. ditto others.

File: src/main/java/bio/terra/common/sam/exception/SamExceptionFactory.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      }
+    }
+    if (!StringUtils.isEmpty(messagePrefix)) {
+      message = messagePrefix + ": " + message;
+    }
+    switch (apiException.getCode()) {
+      case HttpStatusCodes.STATUS_CODE_BAD_REQUEST:
+        {
+          return new SamBadRequestException(message, apiException);
+        }
```
Comment (wchamber) [inline src/main/java/bio/terra/common/sam/exception/SamExceptionFactory.java]: do we need the brackets for every case?

**Review by wchamber — APPROVED** [review]

**Review by ddietterich — APPROVED** [review]
LGTM

File: src/main/java/bio/terra/common/sam/exception/SamBadRequestException.java (hunk 1 above)
Comment (tnadler) [reply src/main/java/bio/terra/common/sam/exception/SamBadRequestException.java:9]: Done

File: src/main/java/bio/terra/common/sam/exception/SamExceptionFactory.java (hunk 2 above)
Comment (tnadler) [reply src/main/java/bio/terra/common/sam/exception/SamExceptionFactory.java]: Done

**Pull request merged**